web/src/compose.js

```diff
@@ -238,14 +247,23 @@ export function send_message(request = create_message_object()) {
             // (Restoring this state is handled by clear_compose_box
             // for locally echoed messages.)
             compose_ui.hide_compose_spinner();
-            return;
-        }
-
-        echo.message_send_error(message.id, response);
+        } else {
+            echo.message_send_error(message.id, response);
 
-        // We might not have updated the draft count because we assumed the
-        // message would send. Ensure that the displayed count is correct.
-        drafts.sync_count();
+            // We might not have updated the draft count because we assumed the
+            // message would send. Ensure that the displayed count is correct.
+            drafts.sync_count();
+        }
+        if (will_split_message) {
+            if (rest_of_the_content) {
```

Comment:
Just struck me... isn't this always true when `will_split_message` is true? Is the else code ever run at all?

Reply:
It's a remnant of the previous logic and is now redundant. I've removed it